### PR TITLE
Bot Management reference architecture diagram

### PR DESCRIPTION
**Pull request by securitypedant**
### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [ ] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- **Log Explorer:**
+  Customers can use Log Explorer to pull additional detailed log data. Logs can easily be filtered by bot score and other bot/traffic/request criteria.
+- **Log Push:**
+  Customers can also export logs to a 3rd party SIEM or analytics platform. Bot score, bot score source, bot detection IDs, and bot detection tags can all be exported as part of the logs.
+
+  ## Bot Management Traffic Flow
+
+![Figure 1: How Cloudflare identifies, scores and processes traffic from bots.](~/assets/images/reference-architecture/bot-management/bot-management-ra-diagram.svg "Figure 1: How Cloudflare identifies, scores and processes traffic from bots.")
+
+1. Client request is sent to the closest Cloudflare Data Center via Anycast ensuring low latency.
```
[the remark this answers is not in the document]
Comment (securitypedant) [reply src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]: @crwaters16 do you think Anycast should be lowercase?

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - Bots
+sidebar:
+  order: 1
+  label: Bot management
+updated: 2024-10-04
+---
+
+## Introduction
+
+Cloudflare has bot management capabilities to help identify and mitigate automated traffic to protect domains from bad bots. [Bot Fight Mode](/bots/get-started/free/) and [Super Bot Fight Mode](/bots/get-started/biz-and-ent/) are options available on free and pro/business accounts respectively. They offer a subset of features/capabilities available for enterprise accounts. This reference architecture diagram focuses on [Enterprise Bot Management](/bots/get-started/bm-subscription/) available for enterprise customers.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
Cloudflare has bot management capabilities to help identify and mitigate automated traffic to protect domains from bad bots. [Bot Fight Mode](/bots/get-started/free/) and [Super Bot Fight Mode](/bots/get-started/biz-and-ent/) are options available on Free and Pro/Business accounts respectively. They offer a subset of features and capabilities available for Enterprise accounts. This reference architecture diagram focuses on [Enterprise Bot Management](/bots/get-started/bm-subscription/) available for Enterprise customers.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+With [Enterprise Bot Management](/bots/get-started/bm-subscription/) customers have the maximum protection and features/capability. A [bot score](/bots/concepts/bot-score/) is exposed for every request. Scores below 30 are commonly associated with bot traffic and customers can then take action on this score with [WAF custom rules](/waf/custom-rules/) or [Workers](/workers/runtime-apis/request/#incomingrequestcfproperties). Additionally, customers can view this score along with other bot specifics like bot score source, bot detection IDs, and bot detection tags in the Bots, Security Analytics, and Events dashboards; these fields can also be seen in more detailed logs in Log Explorer or, with Log Push, logs with these respective fields can be exported to 3rd party SIEMs/Analytics platforms.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
With [Enterprise Bot Management](/bots/get-started/bm-subscription/) customers have the maximum protection, and features and capabilities. A [bot score](/bots/concepts/bot-score/) is exposed for every request. Scores below 30 are commonly associated with bot traffic. Customers can take action on this score by using [WAF custom rules](/waf/custom-rules/) or [Workers](/workers/runtime-apis/request/#incomingrequestcfproperties). Additionally, customers can view this score along with other bot specifics like bot score source, [bot detection IDs](/bots/concepts/detection-ids/), and [bot detection tags](/bots/concepts/detection-ids/#detection-tags) in the Bots, Security Analytics, and Events dashboards. These fields can also be seen in more detailed logs in Log Explorer or exported to third party SIEMs or Analytics platforms using Log Push.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+Cloudflare applies a layered detection approach to Bot Management with several detection engines that cumulatively can impact the bot score given to every request. A bot score is a score from 1 to 99 that indicates how likely that request came from a bot. This bot score can then be used in WAF custom rules or Workers; it's also exposed in a variety of analytics dashboards and logs.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]: This paragraph is a bit repetitive, detailing the same information as the previous paragraph.

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+## Definitions
+
+- **Bot Score:** A [bot score](/bots/concepts/cloudflare-bot-tags/) is a score from 1 to 99 that indicates how likely that request came from a bot. A score of 1 means Cloudflare is quite certain the request was automated.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **Bot Score:** A [bot score](/bots/concepts/cloudflare-bot-tags/) is a score from 1 to 99 that indicates how likely that request came from a bot. A score of 1 means Cloudflare is certain the request was automated.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+- **Bot Score Source:** Bot Score Source is the detection engine used for the bot score.
+- **Bot Detection ID:** [Detection IDs](/bots/concepts/cloudflare-bot-tags/) are static rules used to detect predictable bot behavior with no overlap with human traffic. Detection IDs cause a bot to receive a score source of heuristics with a score of 1\.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **Bot Detection ID:** [Detection IDs](/bots/concepts/detection-ids/) are static rules used to detect predictable bot behavior with no overlap with human traffic. Detection IDs cause a bot to receive a score source of heuristics with a score of 1.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+- **Bot Tag:** [Bot tags](/bots/concepts/cloudflare-bot-tags/) provide more detail about why Cloudflare assigned a [bot score](/bots/concepts/bot-score/) to a request.
+- **Verified Bots:** Cloudflare maintains [a list of "Verified" good bots](https://radar.cloudflare.com/traffic/verified-bots) which can be used in policies to insure good bots such as those associated with a search engine are not blocked.
+- **AI Bots:** [If the feature is enabled](/bots/concepts/bot/), Cloudflare will detect and block verified AI bots that respect robots.txt and crawl rate, and do not hide their behavior from your website. The rule has also been expanded to include more signatures of AI bots that do not follow the rules.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **AI Bots:** [If the feature is enabled](/bots/concepts/bot/#ai-bots), Cloudflare will detect and block verified AI bots that respect `robots.txt` and crawl rate, and do not hide their behavior from your website. The rule has also been expanded to include more signatures of AI bots that do not follow the rules.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+## Cloudflare Bot Management Detection Engines
+
+- **Heuristics:** Cloudflare conducts a number of heuristic checks to identify automated traffic, and requests are matched against a growing database of malicious fingerprints. The Heuristics engine immediately gives automated requests a score of 1\.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **Heuristics:** Cloudflare conducts a number of heuristic checks to identify automated traffic, and requests are matched against a growing database of malicious fingerprints. The [Heuristics engine](/bots/concepts/bot-score/#heuristics) immediately gives automated requests a score of 1.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+- **Machine Learning (ML):** The ML engine accounts for the majority of all detections, human and bot. The ML model leverages Cloudflare's global network, which proxies billions of requests daily, to identify both automated and human traffic. The ML engine produces scores 2 through 99.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **Machine Learning (ML):** The [ML engine](/bots/concepts/bot-score/#machine-learning) accounts for the majority of all detections, human and bot. The ML model leverages Cloudflare's global network, which proxies billions of requests daily, to identify both automated and human traffic. The ML engine produces scores 2 through 99.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+- **Anomaly Detection (AD):** The AD engine is an optional detection engine that uses a form of unsupervised learning. Cloudflare records a baseline of a domain's traffic and uses the baseline to intelligently detect outlier requests. Anomaly Detection is user agent-agnostic and can be turned on or off by your account team. Cloudflare does not recommend AD for domains that use [Cloudflare for SaaS](/cloudflare-for-platforms/cloudflare-for-saas/security/certificate-management/) or expect large amounts of API traffic. The AD engine immediately gives automated requests a score of one.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **Anomaly Detection (AD):** The [AD engine](/bots/concepts/bot-score/#anomaly-detection) is an optional detection engine that uses a form of unsupervised learning. Cloudflare records a baseline of a domain's traffic and uses the baseline to intelligently detect outlier requests. Anomaly Detection is user agent-agnostic and can be turned on or off by your account team. Cloudflare does not recommend AD for domains that use [Cloudflare for SaaS](/cloudflare-for-platforms/cloudflare-for-saas/security/certificate-management/) or expect large amounts of API traffic. The AD engine immediately gives automated requests a score of 1.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+- [**JavaScript Detections (JSD)**](/bots/reference/javascript-detections/)**:** The JSD engine identifies headless browsers and other malicious fingerprints. This engine performs a lightweight, invisible JavaScript injection on the client side of any request. The JSD engine either blocks, challenges, or passes requests to other engines. JSD is enabled by default but completely optional.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **JavaScript Detections (JSD)**: The [JSD engine](/bots/concepts/bot-score/#javascript-detections) identifies headless browsers and other malicious fingerprints. This engine performs a lightweight, invisible JavaScript injection on the client side of any request. The JSD engine either blocks, challenges, or passes requests to other engines. JSD is enabled by default but is completely optional.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+## Bot Dashboards, Analytics, and Logs
+
+Cloudflare bot score and bot traffic analysis is available in several locations.
+
+- **Bots Dashboard:**
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- **Bots dashboard:**
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+  Customers can easily see bot activity up to 30 days back and filter on bot score and other bot/traffic/request filters. The bot feedback loop allows customers to report back to Cloudflare any false positives or false negatives for further investigation.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
  Customers can easily see bot activity up to 30 days back and filter on bot score and other bot, traffic, and request filters. The [bot feedback loop](/bots/concepts/feedback-loop/) allows customers to report back to Cloudflare any false positives or false negatives for further investigation.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+- **Security Analytics:**
+  Security Analytics brings together all of Cloudflare's detection capabilities in one dashboard and provides a broad view of all traffic across the site. The Bots Likelihood graph and widget provide visibility and allow customers to easily view and filter based on bot score and respective categorization of Automated, Likely Automated, Human, and Likely Human.
+- **Events:**
+  Events displays all events the WAF took action on. Events/logs can easily be filtered by bot score and other bot/traffic/request criteria.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
  Events displays all events the WAF took action on. Events and logs can easily be filtered by bot score and other bot, traffic, or request criteria.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+- **Log Explorer:**
+  Customers can use Log Explorer to pull additional detailed log data. Logs can easily be filtered by bot score and other bot/traffic/request criteria.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
  Customers can use Log Explorer to pull additional detailed log data. Logs can easily be filtered by bot score and other bot, traffic, or request criteria.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+- **Log Push:**
+  Customers can also export logs to a 3rd party SIEM or analytics platform. Bot score, bot score source, bot detection IDs, and bot detection tags can all be exported as part of the logs.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
  Customers can also export logs to a third party SIEM or Analytics platform. Bot score, bot score source, bot detection IDs, and bot detection tags can all be exported as part of the logs.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+
+  ## Bot Management Traffic Flow
+
+![Figure 1: How Cloudflare identifies, scores and processes traffic from bots.](~/assets/images/reference-architecture/bot-management/bot-management-ra-diagram.svg "Figure 1: How Cloudflare identifies, scores and processes traffic from bots.")
+
+1. Client request is sent to the closest Cloudflare Data Center via anycast ensuring low latency.
+2. Cloudflare applies a layered approach for bot detection; each detection mechanism impacts the bot score assigned by Cloudflare to every request. Every request is assigned a bot score between 1 \- 99 inclusive.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
2. Cloudflare applies a layered approach for bot detection; each detection mechanism impacts the bot score assigned by Cloudflare to every request. Every request is assigned a bot score between 1-99 inclusive.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+3. Once the client request has been processed by all of Cloudflare's detection engines and assigned a bot score, defined security policies will be executed, some of which may also be leveraging bot score. Various actions can be taken based on the assigned bot score including the block, allow, rate limit, NoCAPTCHA Challenge.
+4. Cloudflare provides analytics and insights into traffic/requests traversing the Cloudflare network. Customers can use the Bots, Security Analytics, Events, and Log Explorer dashboards to understand the overall traffic and bots activity across their site. Customers can also export logs to 3rd party SIEM and Analytics Platforms.
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
4. Cloudflare provides analytics and insights into traffic and requests traversing the Cloudflare network. Customers can use the Bots, Security Analytics, Events, and Log Explorer dashboards to understand the overall traffic and bots activity across their site. Customers can also export logs to third party SIEM and Analytics Platforms.
```

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+
+# Related Resources
+
+- [Cloudflare Bot Management Product Page](https://www.cloudflare.com/application-services/products/bot-management/)
+- [Cloudflare Blog \- Bot Management](https://blog.cloudflare.com/tag/bot-management/)
+- [Cloudflare Developer Doc \- Bots](/bots/)
+- [Video: Cloudflare Bot Management and Turnstile with Demo](https://youtu.be/6EnekTohO7I?si=tk8FUB0xtk1PxsJV)
```
Comment (patriciasantaana) [inline src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]:
```suggestion
- [Cloudflare Bot Management Product Page](https://www.cloudflare.com/application-services/products/bot-management/)
- [Cloudflare Blog - Bot Management](https://blog.cloudflare.com/tag/bot-management/)
- [Bots documentation](/bots/)
- [Video: Cloudflare Bot Management and Turnstile with Demo](https://youtu.be/6EnekTohO7I?si=tk8FUB0xtk1PxsJV)
```

**Review by patriciasantaana — CHANGES_REQUESTED** [review]
A few styling updates + content edits

File: src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx (hunk 4 above)
Comment (securitypedant) [reply src/content/docs/reference-architecture/diagrams/bots/bot-management.mdx]: Ok, the original author, Humair, has updated this section based on your feedback.

**Comment by securitypedant** [comment]
> A few styling updates + content edits

Ok, all changes have been reviewed and commited.

The PR still has Claire as a reviewer and she's out till later in the month. Any chance we can approve this before she gets back?

**Review by patriciasantaana — APPROVED** [review]

**Pull request merged**